.github/workflows/test.yaml

```diff
@@ -14,29 +17,30 @@ jobs:
       - name: Install Erlang & Elixir
         uses: erlef/setup-beam@v1
         with:
-          otp-version: '24.0'
-          elixir-version: '1.13.4'
+          otp-version: ${{ env.otp }}
+          elixir-version: ${{ env.elixir }}
       - name: Cache Mix
         uses: actions/cache@v2
         with:
           path: |
             deps
             _build
-          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          key: ${{ runner.os }}-mix-${{ env.elixir }}-${{ env.otp }}-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            ${{ runner.os }}-mix-
+            ${{ runner.os }}-mix-${{ env.elixir }}-${{ env.otp }}-
       - name: Install mix dependencies
         run: mix deps.get
       - name: Check formatting
-        run: mix format --check-formatted
+        # TODO: remove deps.compile on Elixir > 1.14.0-rc.0
```

Comment:
wait this is being addressed on Elixir main? In any case should we just move it after compile warnings as errors and remove todo? Probably more important to fail on warnings than bad formatting. If you agree Im happy to make the change.

Reply:
It's something else, when there are formatter plugins (heex in our case) running `mix format` should compile the dependencies. Currently the app would be compiled, but not the dependencies, so it would result in seemingly random errors about missing modules from dependencies. It is fixed on elixir main (https://github.com/elixir-lang/elixir/commit/7d2c240eae13cca5869d1f345603eec833dae24d). Does that make sense?

Reply:
Yup, thanks! Still should we swap this with the next command, mix compile —-warnings? Then deps would already have been compiled. I don’t see a downside and it would be one todo less.

Reply:
Ah I see! I'm pretty sure we generally check formatting first on CI in other places, but your reasoning makes total sense. Maybe it's because unless there are plugins we don't need to compile anything to check formatting, hence it should complete faster?

Reply:
Up to you, it is fine to just leave this as is of course.